build/lib/src/asset/exceptions.dart

```diff
@@ -32,13 +32,39 @@ class InvalidOutputException implements Exception {
 }
 
 class InvalidInputException implements Exception {
+  /// The invalid asset that couldn't be read.
   final AssetId assetId;
 
-  InvalidInputException(this.assetId);
+  /// The list of readable globs in the package dependency.
+  ///
+  /// This includes public files like `lib/` by default, but a package can
+  /// choose to allow additional assets via a `build.yaml` file.
+  final List<String> allowedGlobs;
+
+  InvalidInputException(this.assetId, [this.allowedGlobs = const ['lib/**']]);
```

Comment:
nit: Change to a named optional argument (more extensible in the future)

```suggestion
  InvalidInputException(this.assetId, [this.allowedGlobs = const ['lib/**']]);
```